src/logic/wallets/store/reducer/index.ts

```diff
@@ -52,7 +54,13 @@ const providerReducer = handleActions<ProvidersState, ProviderPayloads>(
     [PROVIDER_ACTIONS.NETWORK]: (state: ProvidersState, { payload }: Action<ProviderNetworkPayload>) =>
       providerFactory({ ...state, network: payload }),
     [PROVIDER_ACTIONS.ACCOUNT]: (state: ProvidersState, { payload }: Action<ProviderAccountPayload>) =>
-      providerFactory({ ...state, account: checksumAddress(payload), available: !!payload }),
+      providerFactory({
+        ...state,
+        account: payload ? checksumAddress(payload) : '',
+        available: !!payload,
+      }),
+    [PROVIDER_ACTIONS.SMART_CONTRACT]: (state: ProvidersState, { payload }: Action<ProviderSmartContractPayload>) =>
+      providerFactory({ ...state, smartContractWallet: payload }),
```

Comment:
As discussed on Slack, I'd suggest to remove `smartContractWallet` from the store, and call it dynamically (memoizing if needed).
https://github.com/gnosis/safe-react/blob/dev/src/logic/safe/store/actions/createTransaction.ts#L217-L220 – this is one place where it's already called dynamically. The second place where it's needed is TxModalWrapper.

Reply:
I would suggest we merge this to fix the bug as that is out of scope. I have created a new branch, based off this, to implement that.